File: plugins/afore/src/main/java/eu/geekhome/aforeplugin/AforeAdapterFactory.java
package eu.geekhome.aforeplugin;

import eu.geekhome.services.hardware.HardwareAdapter;
import eu.geekhome.services.hardware.HardwareAdapterFactory;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

class AforeAdapterFactory implements HardwareAdapterFactory {

    @NotNull
    public static final String ID = "AFORE";

    @Override
    public List<HardwareAdapter> createAdapters() {
        ArrayList<HardwareAdapter> result = new ArrayList<>();
        AforeAdapter adapter = new AforeAdapter();
        result.add(adapter);

        return result;
    }

    @Override
    public String getId() {
        return ID;
    }
}

File: plugins/afore/src/main/java/eu/geekhome/aforeplugin/AforeAdapterFactory.kt
package eu.geekhome.aforeplugin

import eu.geekhome.domain.hardware.HardwareAdapter
import eu.geekhome.domain.hardware.HardwareAdapterFactory
import eu.geekhome.domain.langateway.LanGatewayResolver

internal class AforeAdapterFactory(
    override val owningPluginId: String,
    private val lanGatewayResolver: LanGatewayResolver) : HardwareAdapterFactory {

    override fun createAdapters(): List<HardwareAdapter> {
        val result = ArrayList<HardwareAdapter>()
        val adapter = AforeAdapter(owningPluginId, lanGatewayResolver)
        result.add(adapter)
        return result
    }
}